staging/src/k8s.io/client-go/transport/cache.go

```diff
@@ -88,5 +88,5 @@ func tlsConfigKey(c *Config) (string, error) {
 		return "", err
 	}
 	// Only include the things that actually affect the tls.Config
-	return fmt.Sprintf("%v/%x/%x/%x", c.TLS.Insecure, c.TLS.CAData, c.TLS.CertData, c.TLS.KeyData), nil
+	return fmt.Sprintf("%v/%x/%x/%x/%v", c.TLS.Insecure, c.TLS.CAData, c.TLS.CertData, c.TLS.KeyData, c.TLS.ServerName), nil
```

Comment:
Uh, are we printing very sensitive key data into the cache key? That seems like a terrible idea, doesn't it?

Reply:
It's safer to put things into a key struct, that way you don't have to worry about collisions (things maybe having `/` in them that shouldn't, say).

Reply:
> Uh, are we printing very sensitive key data into the cache key? That seems like a terrible idea, doesn't it?

If you have memory access to the cache keys, you also have it to the cache data, which has the same information in a different form

> It's safer to put things into a key struct, that way you don't have to worry about collisions (things maybe having / in them that shouldn't, say).

`/` isn't an issue in this particular case with the current data (`%x` encodes, and the first value is a bool), but agree a key struct would be better here

Reply:
@liggitt 

> If you have memory access to the cache keys, you also have it to the cache data, which has the same information in a different form

I can easily see someone printing the keys to a debug log or something, which is readable by more people than they expect. People don't usually expect cache keys to be super sensitive.

I agree the separator is likely not an actual bug *in this particular case*, but if you use the struct approach everywhere you prevent an entire class of really hard to find bugs, some of which are serious security problems.

I will likely file an issue about these items.

Reply:
fair enough. opened https://github.com/kubernetes/kubernetes/pull/56811 to switch to struct key with its own String() impl that omits logging the key data

Reply:
Thanks!